### PR TITLE
WIP: Feature/#727 add guided dialog for adding a whitelist

### LINKED ISSUE #509: See all errors by uploading an obs-csv-file
# User Story

### As a: 
Teacher

### I want:
to upoad a csv file for whitelists and see all errors in an error message in one view. 

### so that:
I know which parts of my file are not correct.

## Acceptance criteria:
- [ ] Collect all errors of a csv-file

## Additional info:

------
_Please tag this issue if you are sure to which tag(s) it belongs._

### LINKED ISSUE #727: Add guided dialog for adding a whitelist
# User Story

### I want: a guided dialog for adding a whitelist. 
Currently the feature is not that good, in a UX aspect. The dialog should be guided. So the screen is for uploading a file. The second screen should show the user, which are in geli and which are currently not. Also a accept or cancel button should be there. The functionality is present but the UI is not that good.

## Acceptance criteria:
- [ ] new dialog for whitelist feature
- [ ] use the shared upload component
- [ ] show the user at the end of the dialog
- [ ] don't create the whitelist user object in the db until the user accept

## Estimated:
- 5 x 4hr

### PR DESCRIPTION
**Pull request by lukas-schardt**
## Description:
Adds a guided dialog for creating a whitelist

Closes #509, closes #727 

## Improvements
- 

## Known Issues:
_NONE_

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -39,7 +39,6 @@
     "body-parser": "^1.18.2",
     "express": "^4.16.3",
     "fast-csv": "^2.4.1",
-    "fs-extra": "^5.0.0",
```
Comment (danielkesselberg) [inline api/package.json]: @PatrickSkowronek i removed fs-extra as dependency a few weeks ago but with another feature it came back to geli. I could not find any reference with current develop. Do you think its save to remove it?

File: app/webFrontend/src/app/course/course-edit/general-tab/whitelist-dialog/whitelist-dialog.component.html [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    <mat-step>
+
+      <ng-template matStepLabel>{{ 'course.text.removeUnwantedUsers' | translate}}</ng-template>
+
+      <ng-container *ngFor="let user of whitelistUsers; let i = index;">
+        <mat-card>
+          <ul class="info">
+
+            <li class="existing">
+              <mat-icon *ngIf="!user.exists" color="warn" matTooltip="{{ 'user.doesntexist' | translate }}">warning</mat-icon>
```
[the remark this answers is not in the document]
Comment (danielkesselberg) [reply app/webFrontend/src/app/course/course-edit/general-tab/whitelist-dialog/whitelist-dialog.component.html]: @lukas-schardt i added the missing string to de.json and renamed key.

File: api/src/controllers/WhitelistController.ts [hunk 3]
```diff
@@ -11,11 +11,64 @@ import ObjectId = mongoose.Types.ObjectId;
...
+   *        "exists": true
+   *    },
+   *    {
+   *        "uid": "<non-existing-uid>",
+   *        "exists": false
+   *    }
+   *  ]
+   * @param data
+   * @param currentUser
+   * @param whitelistToCheck
```
Comment (danielkesselberg) [inline api/src/controllers/WhitelistController.ts]: Please change `@param` to `@apiParam`

File: api/package.json (hunk 1 above)
Comment (PatrickSkowronek) [reply api/package.json]: yeah, I think it is a boomerang. I removed it in other branches, but it's still coming back.

**Pull request merged**